### PR TITLE
Azure ml service

### PR DESCRIPTION
**Pull request by Blackmist**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Incorporated changes suggested by bmoore-msft
* Removed output as it just returns the workspace name, which is just what you entered originally

### CODE REVIEW

**Comment by Blackmist** [comment]
@bmoore-msft No luck after merging the new .travis.yml. Same error.

File: 101-azure-machine-learning-service-workspace/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Azure Machine Learning service workspace
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/README.md]: use title casing

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    "parameters": {
+        "workspaceName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the Azure Machine Learning service workspace."
+            }
+        },
+        "location": {
+            "type": "string",
+            "allowedValues": [
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: remove the spaces from allowedValues and make all lower case

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                    "family": "A"
+                },
+                "accessPolicies": []
+            }
+        },
+        {
+            "name": "[variables('applicationInsightsName')]",
+            "type": "Microsoft.Insights/components",
+            "apiVersion": "2015-05-01",
+            "location": "[parameters('location')]",
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: This needs to be restricted to certain locations - will need another param if it doesn't match your current restriction

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            "type": "Microsoft.MachineLearningServices/workspaces",
+            "apiVersion": "2018-11-19",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[variables('storageAccount').name]",
+                "[variables('keyVault').name]",
+                "[variables('applicationInsightsName')]",
+                "[variables('containerRegistryName')]"
+            ],
+            "identity": {
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: how is this used?

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the Azure Machine Learning service workspace."
+            }
+        },
+        "location": {
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: looks like a number of resources in this have limited location availability - need to set allowed values for the union of those.

File: 101-azure-machine-learning-service-workspace/metadata.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/metadata.json]: format the json (e.g. SHIFT+ALT+F in VS Code)

File: 101-azure-machine-learning-service-workspace/azuredeploy.parameters.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.parameters.json]: format the json (e.g. SHIFT+ALT+F in VS Code)

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 2 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: Done

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 3 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: These should match; when you create through the portal everything gets created in the same locations. What's supported by Azure Machine Learning service should be the intersection of what all the other dependencies allow.

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 5 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: When the workspace is created, a managed system identity is created and associated with it. When the workspace is deleted, the managed system identity is also removed. It's required by the workspace.

File: 101-azure-machine-learning-service-workspace/metadata.json (hunk 6 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/metadata.json]: done, though some files have spacing of 4 and one has a spacing of 2. Does that matter?

File: 101-azure-machine-learning-service-workspace/azuredeploy.parameters.json (hunk 7 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.parameters.json]: done

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 4 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: When the workspace is created, a managed system identity is created and associated with it. When the workspace is deleted, the managed system identity is also removed. It's required by the workspace.

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 5 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: Apparently the advice I got on that was incorrect. Checking now, it looks like some of the resources are allowed outside of the region. Specifically looks like app insights. Adding another regional list for that.

Is this something that we can set automatically? Or will we have to prompt the user for the location of each resource?

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 3 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: What about an IF statement? There's only two locations that aren't available, east us 2 and west central us. Using an IF, I can split the difference and select south central us for both of those for the app insight resource.

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 5 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: What about an IF statement? There's only two locations that aren't available, east us 2 and west central us. Using an IF, I can split the difference and select south central us for both of those for the app insight resource.

**Pull request closed without merging**